pkg/kubectl/cmd/util/clientcache.go

```diff
@@ -119,15 +119,18 @@ func (c *ClientCache) clientConfigForVersion(requiredVersion *schema.GroupVersio
 		return copyConfig(c.noVersionConfig), nil
 	}
 
+	// TODO this isn't what we want.  Each clientset should be setting defaults as it sees fit.
+	oldclient.SetKubernetesDefaults(&config)
+	if requiredVersion == nil {
+		requiredVersion = config.GroupVersion
```

Comment:
That's the intention, the old defaults always set the GV to `/v1`, which is the previous behavior, but additionally if `requiredVersion` is set, we'll override the default with that.